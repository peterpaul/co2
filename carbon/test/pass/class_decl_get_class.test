/*
 * carbon Compiler and programming language for libco2
 *
 * Copyright (C) 2011,2012 Peterpaul Taekele Klein Haneveld
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
["co2/Object.h"] {
	class Object {
		Class class;
	}

	class Class : Object {
		char * class_name;
	}
}

["stdio.h"] {
	int printf(char *format, ...);
}

class MyObject : Object {
	void print () {
		printf ("class.class_name = %s\n", class.class_name);
	}
}

int main() {
	MyObject o = new MyObject ();
	o.print ();
	printf ("o.class.class_name = %s\n", o.class.class_name);
	return 0;
}
