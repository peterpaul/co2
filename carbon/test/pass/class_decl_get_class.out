class.name = MyObject
o.class.name = MyObject
